nixos/modules/programs/zsh/zsh.nix

```diff
@@ -97,45 +95,6 @@ in
 
   config = mkIf cfg.enable {
 
-    programs.zsh = {
-
-      shellInit = ''
-        . ${config.system.build.setEnvironment}
-
-        ${cfge.shellInit}
-      '';
-
-      loginShellInit = cfge.loginShellInit;
-
-      interactiveShellInit = ''
-        # history defaults
-        SAVEHIST=2000
-        HISTSIZE=2000
-        HISTFILE=$HOME/.zsh_history
-
-        setopt HIST_IGNORE_DUPS SHARE_HISTORY HIST_FCNTL_LOCK
-
-        # Tell zsh how to find installed completions
-        for p in ''${(z)NIX_PROFILES}; do
-          fpath+=($p/share/zsh/site-functions $p/share/zsh/$ZSH_VERSION/functions)
-        done
-
-        ${if cfg.enableCompletion then "autoload -U compinit && compinit" else ""}
```

Comment:
Actually it's the other way around. Auto-loaded may use `compadd` which require completion to be enable. There is no direct relationship between `compinit` and `fpath`. Therefore I will move `compinit`up.